### PR TITLE
feat: MCP OAuth - controller wiring and app integration (4/7)

### PR DESCRIPTION
**Pull request by hanneskuettner**
## Scope

This is PR 4/7 for MCP OAuth support. It wires the service from PR 3 into HTTP routes, app mounting, login redirect handling, and integration-style tests.

This PR adds the public and protected MCP OAuth routers:

- discovery metadata endpoints
- `/mcp-oauth/authorize`
- `/mcp-oauth/authorize/decision`
- `/mcp-oauth/register`
- `/mcp-oauth/token`
- `/mcp-oauth/revoke`

It also mounts the guard in `app.ts`, updates login redirect handling for OAuth authorization redirects, and adds controller/schedule coverage around the wired routes.

Review fixes now included here:

- non-`OAuthError` values in the authorize route are rethrown to the normal Directus error handler
- disabled token responses include `Cache-Control: no-store` and CORS headers where needed
- settings reads use `SettingsService`
- redirect matching compares the full normalized URI, including query string
- login redirects reject backslashes to avoid browser URL normalization open redirects
- cleanup schedule tests mock the correct service path and execute the scheduled callback

## Potential Risks / Drawbacks

- The authorize route intentionally has browser behavior: missing/invalid sessions redirect to login, trusted OAuth validation errors redirect to the client, and pre-trust OAuth errors render a local error page.
- Re-throwing non-OAuth errors is a behavior change from local 400 rendering, but it is the safer Directus-native behavior for unexpected failures.

## Tested Scenarios

- Controller responses for discovery, authorize, decision, token, register, and revoke paths
- disabled settings gate responses and headers
- cleanup scheduled callback execution and error logging
- redirect matching edge cases
- login redirect open-redirect protection

The final restacked branch was also verified with the MCP OAuth service/schedule tests and `@directus/api` build.

## Review Notes / Questions

- The public OAuth router remains before regular authentication because discovery, DCR, token, and revoke are OAuth protocol endpoints. The consent decision route remains protected by cookie auth and same-origin checks.

## Checklist

- [x] Added or updated tests
- [ ] Documentation PR created [here](https://github.com/directus/docs) or not required
- [ ] OpenAPI package PR created [here](https://github.com/directus/openapi) or not required

### CODE REVIEW

File: api/src/controllers/mcp-oauth.ts [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+			const consentData: ConsentPageData = {
+				clientName: result.client_name,
+				redirectUri: result.redirect_uri,
+				scope: result.scope,
+				signedParams: result.signed_params,
+				decisionUrl,
+			};
+
+			res.send(await renderConsentPage(consentData, pageOpts));
+		} catch (err) {
```
Comment (bryantgillespie) [inline api/src/controllers/mcp-oauth.ts:393]: Should we re-throw here for non OAuthErrors?

**Comment by hanneskuettner** [comment]
Follow-up PR for settings toggle + client listing UI: #27105

File: api/src/services/mcp-oauth/utils/redirect.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export function matchRedirectUri(requested: string, registered: string[]): boolean {
+	return registered.some((reg) => {
+		if (reg === requested) return true;
+
+		try {
+			const regUrl = new URL(reg);
+			const reqUrl = new URL(requested);
+
+			if (isLoopbackHost(regUrl.hostname)) {
```
Comment (bryantgillespie) [inline api/src/services/mcp-oauth/utils/redirect.ts:76]: Should probably compare more than just hostname protocol and pathname here yeah?

File: app/src/routes/login/utils/navigate-after-login.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { getRootPath } from '@/utils/get-root-path';
+
+/**
+ * Navigate to the appropriate page after login.
+ * API routes (e.g., /mcp-oauth/authorize) need full page navigation
+ * since they're not SPA routes.
+ */
+export function navigateAfterLogin(router: Router, target: string): void {
+	// Reject non-relative paths to prevent open redirect
+	if (!target.startsWith('/') || target.startsWith('//')) {
```
Comment (bryantgillespie) [inline app/src/routes/login/utils/navigate-after-login.ts]: One from AI friends

----

Open redirect: `/\evil.com` bypasses the `//` guard. WHATWG URL spec normalizes `\` → `/` in http(s), so `new URL('/\evil.com', origin)` → `evil.com` in Chrome/Firefox/Safari/Node.

  ```suggestion
        // Reject non-relative paths to prevent open redirect.
        if (!target.startsWith('/') || target.startsWith('//') || target.includes('\\')) {
                router.push('/');
                return;
        }
  ```

File: api/src/schedules/oauth-cleanup.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { useEnv } from '@directus/env';
 import { useLogger } from '../logger/index.js';
-import { McpOAuthService } from '../services/mcp-oauth.js';
+import { McpOAuthService } from '../services/mcp-oauth/index.js';
```
Comment (bryantgillespie) [inline api/src/schedules/oauth-cleanup.ts:3]: Looks like this test is still mocking the old service path (`../services/mcp-oauth.js`) even though the schedule now imports `../services/mcp-oauth/index.js`).

It also only verifies registration, so it wouldn’t catch a broken cleanup path. Could we capture and run the scheduled callback here and assert `getSchema()` and `cleanup()` happen?
